annotation/src/main/kotlin/com/tobrun/datacompat/annotation/Default.kt

```diff
@@ -0,0 +1,12 @@
+package com.tobrun.datacompat.annotation
+
+/**
+ * Default value represented as a String.
+ *
+ * @param valueAsString exact representation of the default value. E.g. if default [String] is used,
+ *  it should be passed here as "\"STRING_VALUE\""; if default [Int] is used, it should be passed
+ *  as "INT_VALUE".
+ */
+@Retention(AnnotationRetention.RUNTIME)
+@Target(AnnotationTarget.CLASS)
+annotation class Default(val valueAsString: String)
```

Comment:
We couldn't use Any as type? or have "overloads"?

Reply:
Should this annotation be like an nested/inner class? 
Many libraries expose this sub annotations as `DataCompatDefault` or `DataCompat.Default`

Reply:
`Any` is not supported as annotation parameter sadly